source/features/collapse-markdown-sections.ts

```diff
@@ -0,0 +1,68 @@
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+const allHeadingsSelector = 'h1, h2, h3, h4, h5, h6';
+
+function isHeading(element: HTMLElement): boolean {
+	return element.matches(allHeadingsSelector);
+}
+
+function isSameLevelHeading(referenceHeading: HTMLElement, compareHeading: HTMLElement): boolean {
+	return referenceHeading.tagName === compareHeading.tagName;
+}
+
+function isHigherLevelHeading(referenceHeading: HTMLElement, compareHeading: HTMLElement): boolean {
+	return referenceHeading.tagName.localeCompare(compareHeading.tagName) > 0;
+}
+
+function toggleSection(event: delegate.Event<MouseEvent, HTMLElement>): void {
+	let sectionHeading = (event.target as HTMLElement).closest<HTMLElement>(allHeadingsSelector)!;
+	const sectionState = sectionHeading.classList.contains('rgh-markdown-section-collapsed');
+	if (event.altKey) {
+		// Set `sectionHeading` as the first same-level header of the larger section
+		for (let element = sectionHeading; element; element = element.previousElementSibling as HTMLElement) {
+			if (!isHeading(element)) {
+				continue;
+			}
+
+			if (isHigherLevelHeading(sectionHeading, element)) {
+				break;
+			}
+
+			if (isSameLevelHeading(sectionHeading, element)) {
+				sectionHeading = element;
+			}
+		}
+	}
+
+	for (let element = sectionHeading; element; element = element.nextElementSibling as HTMLElement) {
```

Comment:
```suggestion
	let element = sectionHeading;
	while (element) {
```

And then `element = element.nextElementSibling as HTMLElement` at the end.

Same goes for the previous loop